search/connection.py

```diff
@@ -2,62 +2,50 @@
 Elasticsearch connection functionality
 """
 from functools import partial
+import logging
 import uuid
 
 from django.conf import settings
 from elasticsearch_dsl.connections import connections
 
 from search.constants import VALID_OBJECT_TYPES
 
-_CONN = None
-# When we create the connection, check to make sure all appropriate mappings exist
-_CONN_VERIFIED = False
 
+log = logging.getLogger()
 
-def get_conn(*, verify=True):
-    """
-    Lazily create the connection.
-
-    Args:
-        verify (bool): If true, check the presence of indices and mappings
 
-    Returns:
-        elasticsearch.client.Elasticsearch: An Elasticsearch client
+def configure_connections():
     """
-    # pylint: disable=global-statement
-    global _CONN
-    global _CONN_VERIFIED
+    Create connections for the application
 
-    do_verify = False
-    if _CONN is None:
+    This should only be called once
+    """
+    if settings.ELASTICSEARCH_URL:
+        # this is the default connection
         http_auth = settings.ELASTICSEARCH_HTTP_AUTH
         use_ssl = http_auth is not None
-        _CONN = connections.create_connection(
-            hosts=[settings.ELASTICSEARCH_URL],
-            http_auth=http_auth,
-            use_ssl=use_ssl,
-            # make sure we verify SSL certificates (off by default)
-            verify_certs=use_ssl,
+        # configure() lazily creates connections when get_connection() is called
+        connections.configure(
+            default={
+                "hosts": [settings.ELASTICSEARCH_URL],
+                "http_auth": http_auth,
+                "use_ssl": use_ssl,
+                # make sure we verify SSL certificates (off by default)
+                "verify_certs": use_ssl,
+            }
         )
-        # Verify connection on first connect if verify=True.
-        do_verify = verify
+    else:
+        log.error("ELASTICSEARCH_URL is not set")
```

Comment:
Should we raise an exception here? This setting should always be true.

Reply:
I removed the condition here and instead put a check for the setting in `settings.py` so the app will fail to deploy and heroku will roll the app back.